File: moviejava/app/src/main/java/com/example/moviejava/data/remote/genres/GenresRepository.java
package com.example.moviejava.data.remote.genres;


import androidx.annotation.NonNull;

import com.example.moviejava.model.JsonObject;
import com.example.moviejava.model.genres.FromJson;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.Retrofit;

@Singleton
public class GenresRepository {
    @NonNull
    private final GenresService service;

    @Inject
    public GenresRepository(@NonNull Retrofit retrofit) {
       this.service = retrofit.create(GenresService.class);
    }


    public Observable<FromJson> getGenres() {

        return service.getListGenres();
    }

    public Observable<JsonObject> getListGenresByFilm(int id, int page) {

        return service.getListFilmByGenres(id, page);
    }

}

File: movie/app/src/main/java/com/example/movie/data/remote/genres/GenresRepository.kt
package com.example.movie.data.remote.genres

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.flowOn
import retrofit2.Retrofit
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class GenresRepository @Inject constructor(val retrofit: Retrofit) {
    private val service = retrofit.create(GenresService::class.java)
    fun getGenres() = flow {
        emit(
            service.getListGenres()
        )
    }.flowOn(Dispatchers.IO)

    fun getListFilmByGenres(id: Int, page: Int) = flow {
        emit(service.getListFilmByGenres(id = id, page = page))
    }.flowOn(Dispatchers.IO)
}